sdk/data/azcosmos/otel_constants.go

```diff
@@ -0,0 +1,176 @@
+}
+
+func getSpanNameForContainers(endpoint *url.URL, operationType operationType, resourceType resourceType, database string, id string) (span, error) {
+	var spanName string
+	switch resourceType {
+	case resourceTypeCollection:
+		switch operationType {
+		case operationTypeCreate:
+			spanName = otelSpanNameCreateContainer
+		case operationTypeRead:
+			spanName = otelSpanNameReadContainer
+		case operationTypeDelete:
+			spanName = otelSpanNameDeleteContainer
+		case operationTypeReplace:
+			spanName = otelSpanNameReplaceContainer
+		case operationTypeBatch:
+			spanName = otelSpanNameExecuteBatch
+		}
+	case resourceTypeOffer:
+		switch operationType {
+		case operationTypeRead:
+			spanName = otelSpanNameReadThroughputContainer
+		case operationTypeReplace:
+			spanName = otelSpanNameReaplaceThroughputContainer
+		}
+	}
+
+	if spanName == "" {
+		return span{}, fmt.Errorf("undefined telemetry span for operationType %v and resourceType %v", operationType, resourceType)
+	}
+
+	return span{name: fmt.Sprintf("%s %s", spanName, id), options: getSpanPropertiesForContainer(endpoint, spanName, database, id)}, nil
+}
+
+func getSpanNameForItems(endpoint *url.URL, operationType operationType, database string, id string) (span, error) {
+	var spanName string
+	switch operationType {
+	case operationTypeCreate:
+		spanName = otelSpanNameCreateItem
+	case operationTypeRead:
+		spanName = otelSpanNameReadItem
+	case operationTypeDelete:
+		spanName = otelSpanNameDeleteItem
+	case operationTypeReplace:
+		spanName = otelSpanNameReplaceItem
+	case operationTypeUpsert:
+		spanName = otelSpanNameUpsertItem
+	case operationTypePatch:
+		spanName = otelSpanNamePatchItem
+	case operationTypeQuery:
+		spanName = otelSpanNameQueryItems
+	}
+
+	if spanName == "" {
+		return span{}, fmt.Errorf("undefined telemetry span for operationType %v and resourceType %v", operationType, resourceTypeDocument)
+	}
+
+	return span{name: fmt.Sprintf("%s %s", spanName, id), options: getSpanPropertiesForContainer(endpoint, spanName, database, id)}, nil
+}
+
+func getSpanPropertiesForClient(endpoint *url.URL, operationName string) runtime.StartSpanOptions {
+	return runtime.StartSpanOptions{Attributes: []tracing.Attribute{
+		{Key: "db.system", Value: "cosmosdb"},
+		{Key: "db.cosmosdb.connection_mode", Value: "gateway"},
+		{Key: "db.operation.name", Value: operationName},
+		{Key: "server.address", Value: endpoint.Hostname()},
+		{Key: "server.port", Value: endpoint.Port()},
+	}}
+}
+
+func getSpanPropertiesForDatabase(endpoint *url.URL, operationName string, id string) runtime.StartSpanOptions {
+	return runtime.StartSpanOptions{Attributes: []tracing.Attribute{
+		{Key: "db.system", Value: "cosmosdb"},
+		{Key: "db.cosmosdb.connection_mode", Value: "gateway"},
+		{Key: "db.namespace", Value: id},
+		{Key: "db.operation.name", Value: operationName},
+		{Key: "server.address", Value: endpoint.Hostname()},
+		{Key: "server.port", Value: endpoint.Port()},
```

Comment:
WDYT about only setting the port if it's not default (i.e. not 443) ? I assume in absolute majority of cases it is and it's just wasteful to set it.

I'm also proposing to change it here: https://github.com/open-telemetry/semantic-conventions/pull/1330

Reply:
I can do that change, sure